tests/integrations/android_sdk/test_verify_android_sdk.py

```diff
@@ -0,0 +1,281 @@
+    "Generate a side effect method that will accept a license."
+    def _side_effect(*args, **kwargs):
+        license_dir = root_path / "licenses"
+        license_dir.mkdir(parents=True)
+        (license_dir / "android-sdk-license").touch()
+
+    return _side_effect
+
+
+@pytest.mark.parametrize("host_os", ("ArbitraryNotWindows", "Windows"))
+def test_succeeds_immediately_in_happy_path(mock_command, host_os, tmp_path):
+    "If verify_android_sdk is invoked on a valid path, it does nothing."
+    # `sdkmanager` exists and has the right permissions, and
+    # `android-sdk-license` exists, verify_android_sdk() should
+    # succeed, create no subprocesses, make no requests, and return a
+    # SDK wrapper.
+
+    # On Windows, this requires `sdkmanager.bat`; on non-Windows, it requires
+    # `sdkmanager`.
+
+    # Create `sdkmanager` and the license file.
+    root_path = tmp_path / ".briefcase" / "tools" / "android_sdk"
+    tools_bin = root_path / "tools" / "bin"
+    tools_bin.mkdir(parents=True, mode=0o755)
+    if host_os == "Windows":
+        (tools_bin / "sdkmanager.bat").touch()
+    else:
+        (tools_bin / "sdkmanager").touch(mode=0o755)
+    licenses = root_path / "licenses"
+    licenses.mkdir(parents=True, mode=0o755)
+    (licenses / "android-sdk-license").touch()
+
+    # Configure `mock_command` to assume the `host_os` we parameterized with.
+    mock_command.host_os = host_os
+
+    # Expect verify_android_sdk() to succeed
+    sdk = verify_android_sdk(mock_command)
+
+    # No calls to download, run or unpack anything.
+    mock_command.download_url.assert_not_called()
+    mock_command.subprocess.run.assert_not_called()
+    mock_command.subprocess.check_output.assert_not_called()
+    mock_command.shutil.unpack_archive.assert_not_called()
+
+    # The returned SDK has the expected root path.
+    assert sdk.root_path == root_path
+
+
+def test_user_provided_sdk(mock_command, tmp_path):
+    "If the user specifies a valid ANDROID_SDK_ROOT, it is used"
+    # Create `sdkmanager` and the license file.
+    root_path = tmp_path / "other_sdk"
+    tools_bin = root_path / "tools" / "bin"
+    tools_bin.mkdir(parents=True, mode=0o755)
+    (tools_bin / "sdkmanager").touch(mode=0o755)
+    licenses = root_path / "licenses"
+    licenses.mkdir(parents=True, mode=0o755)
+    (licenses / "android-sdk-license").touch()
+
+    # Set the environment to specify ANDROID_SDK_ROOT
+    mock_command.os.environ = {
+        'ANDROID_SDK_ROOT': str(root_path)
+    }
+
+    # Expect verify_android_sdk() to succeed
+    sdk = verify_android_sdk(mock_command)
+
+    # No calls to download, run or unpack anything.
+    mock_command.download_url.assert_not_called()
+    mock_command.subprocess.run.assert_not_called()
+    mock_command.subprocess.check_output.assert_not_called()
+    mock_command.shutil.unpack_archive.assert_not_called()
+
+    # The returned SDK has the expected root path.
+    assert str(sdk.root_path) == str(root_path)
```

Comment:
Line 113 asserts `str(x) == str(y)`, but line 135 asserts on the paths themselves. I propose you stick to asserting the paths are equal.

Reply:
Yeah... so... I completely agree. Python 3.5 doesn't :-) 
```
==============================================================================
FAIL: tests/integrations/android_sdk/test_verify_android_sdk.py::test_user_provided_sdk
------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rkm/projects/beeware/briefcase/tests/integrations/android_sdk/test_verify_android_sdk.py", line 113, in test_user_provided_sdk
    assert sdk.root_path == root_path
AssertionError: assert PosixPath('/private/var/folders/9p/nzv40l_974ddhp1hfrccbpw80000gn/T/pytest-of-rkm/pytest-2/test_user_provided_sdk0/other_sdk') == PosixPath('/private/var/folders/9p/nzv40l_974ddhp1hfrccbpw80000gn/T/pytest-of-rkm/pytest-2/test_user_provided_sdk0/other_sdk')
 +  where PosixPath('/private/var/folders/9p/nzv40l_974ddhp1hfrccbpw80000gn/T/pytest-of-rkm/pytest-2/test_user_provided_sdk0/other_sdk') = <briefcase.integrations.android_sdk.AndroidSDK object at 0x103dce7b8>.root_path

------------------------------------------------------------------------------
```
That is:
```
/private/var/folders/9p/nzv40l_974ddhp1hfrccbpw80000gn/T/pytest-of-rkm/pytest-2/test_user_provided_sdk0/other_sdk
```
!=
```
/private/var/folders/9p/nzv40l_974ddhp1hfrccbpw80000gn/T/pytest-of-rkm/pytest-2/test_user_provided_sdk0/other_sdk
```
This exact test passes under Python 3.6+. I can only assume it's some obscure bug about comparison of paths that is causing the failure.

I'll add a FIXME that can be cleaned up once we drop Python 3.5 support (along with migrating to f-strings instead of `.format()`...)